interfaces/apparmor/backend.go

```diff
@@ -312,17 +312,11 @@ func profileIsRemovableOnCoreSetup(fn string) bool {
 	return true
 }
 
-// Setup creates and loads apparmor profiles specific to a given snap.
-// The snap can be in developer mode to make security violations non-fatal to
-// the offending application process.
-//
-// This method should be called after changing plug, slots, connections between
-// them or application present in the snap.
-func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions, repo *interfaces.Repository, tm timings.Measurer) error {
+func (b *Backend) prepareProfiles(snapInfo *snap.Info, opts interfaces.ConfinementOptions, repo *interfaces.Repository) (changedPaths, unchangedPaths, removedPaths []string, err error) {
```

Comment:
could the return type be simplified here to use a struct like:

```go
type profilePathsResults struct {
    changed []string
    unchanged []string
    removed []string
}
```

that would make it a bit easier to read this and you wouldn't have as many `return nil, nil, nil, err` which looks a bit unwieldy IMHO